awx/ui/client/lib/models/Base.js

```diff
@@ -143,6 +147,19 @@ function httpOptions (resource) {
         });
 }
 
+function httpDelete (resource) {
+    const req = {
+        method: 'DELETE',
+        url: this.path
+    };
+
+    if (resource) {
+        req.url = `${this.path}${resource}/`;
+    }
+
+    return $http(req).then(res => res);
```

Comment:
The `then( ... )` on this line is unneeded since the return value is the same as what `$http(req)` returns.